### PR TITLE
bug fixes in for co2 in both nuopc and mct caps

### PR DESCRIPTION
**Pull request by mvertens**
### Description of changes
This PR fixes the way the way co2 from the atm is handled in both the nuopc and mct caps.

### Specific notes

Contributors other than yourself, if any:

CTSM Issues Fixed (include github issue #):

Are answers expected to change (and if so in what way)?

Any User Interface Changes (namelist or namelist defaults changes)?

Testing performed, if any:
(List what testing you did to show your changes worked as expected)
(This can be manual testing or running of the different test suites)
(Documentation on system testing is here: https://github.com/ESCOMP/ctsm/wiki/System-Testing-Guide)
(aux_clm on cheyenne for intel/gnu and izumi for intel/gnu/nag/pgi is the standard for tags on master)

**NOTE: Be sure to check your coding style against the standard
(https://github.com/ESCOMP/ctsm/wiki/CTSM-coding-guidelines) and review
the list of common problems to watch out for
(https://github.com/ESCOMP/CTSM/wiki/List-of-common-problems).**

### CODE REVIEW

**Review by billsacks — CHANGES_REQUESTED** [review]
Thank you very much for fixing these issues!

One change here looks suspicious - see below.

File: src/cpl/mct/lnd_comp_mct.F90 [hunk 1]
```diff
@@ -247,8 +248,16 @@ subroutine lnd_init_mct( EClock, cdata_l, x2l_l, l2x_l, NLFilename )
...
+#if (defined COMPARE_TO_NUOPC)
+       if (nsrest == nsrStartup) then
+          call ESMF_ClockGet( Eclock, currTime=currTime)
+          call ESMF_TimeGet( currTime, dayOfYear_r8=nextsw_cday)
+       else
+          call seq_infodata_GetData(infodata, nextsw_cday=nextsw_cday )
+       end if
+#else
        call seq_infodata_GetData(infodata, nextsw_cday=nextsw_cday )
-       call set_nextsw_cday(nextsw_cday)
```
[the remark this answers is not in the document]
Comment (mvertens) [reply src/cpl/mct/lnd_comp_mct.F90:251]: So this is really interesting. I removed the entire following block in a test and the results are bfb.

**Comment by billsacks** [comment]
I'm going to split this into two separate tags: one that fixes co2 and one that removes the unnecessary nextsw_cday in initialization. Even though both should appear bfb in testing, I'd like to isolate the co2 fix to its own tag because it is significantly answer changing in coupled configurations, so I want a clean tag with just that fix. I'm also *not* going to merge this with #1319 , for a similar reason. So I'll be making three tags in quick succession over the weekend.

**Review by billsacks — APPROVED** [review]

**Comment by billsacks** [comment]
Okay, on to the second part of this PR: the removal of nextsw_cday from initialization.

I have double-checked and agree with @mvertens that this isn't needed in initialization. My recollection is that she did a big rework of initialization maybe 5 years ago and removed the call to SurfaceAlbedo in initialization at that time, so ever since then we haven't needed nextsw_cday in initialization.

It looks to me like we could now probably do away with nextsw_cday from clm_time_manager entirely, since it is passed directly into clm_driver. But I'm not 100% sure of that, and I'm out of time right now, so I'm going to move ahead with just the changes @mvertens has already implemented.

**Comment by billsacks** [comment]
> It looks to me like we could now probably do away with nextsw_cday from clm_time_manager entirely, since it is passed directly into clm_driver. But I'm not 100% sure of that, and I'm out of time right now, so I'm going to move ahead with just the changes @mvertens has already implemented.

Actually, it only took me a few minutes to confirm this: yes, nextsw_cday is never used in clm_time_manager, so I'm going to remove it from there (also removing the call to set_nextsw_cday) to avoid confusion: this value is passed directly to clm_driver rather than being queried from clm_time_manager.

**Pull request merged**